src/TRestAxionTrueWolterOptics.cxx

```diff
@@ -321,12 +333,12 @@ Int_t TRestAxionTrueWolterOptics::SecondMirrorReflection(const TVector3& pos, co
     TVector3 vertex(0, 0, fBackVertex[mirror]);
     Double_t focal = fR3[mirror] / TMath::Tan(4 * fAlpha[mirror]);
 
-    //// Reflection on first mirror
-    fSecondInteractionPosition =
-        REST_Physics::GetHyperbolicVectorIntersection(pos, dir, fAlpha[mirror], fR3[mirror], fMirrorLength,
-                                                      focal);  // should add this: TVector3(0, 0, -1), vertex,
+    //// Reflection on second mirror
+    fSecondInteractionPosition = REST_Physics::GetHyperbolicVectorIntersection(
+        pos, dir, fAlpha[mirror], fR3[mirror], focal);  // should add this: TVector3(0, 0, -1), vertex,
```

Comment:
Imo it's confusing that this function receives α and you compute 3α in the body of the function. Ideally, have a separate angle that simply describes the correct angle for the second set of mirrors, or compute it in place to clarify? It makes `GetHyperbolicVectorIntersection` much more specific than the name would imply, i.e. it forces it to be explicitly for a second set of mirrors in a Wolter optic.